src/client/bridge/gateway/shard_runner.rs

```diff
@@ -548,8 +559,36 @@ impl ShardRunner {
             Ok(None) => None,
             Err(why) => {
                 error!("Shard handler received err: {:?}", why);
+                match why {
+                    Error::Gateway(crate::prelude::GatewayError::InvalidAuthentication) => {
```

Comment:
Please simplify `crate::prelude::GatewayError::InvalidAuthentication` to just `GatewayError::InvalidAuthentication`.